File: Engineering concepts/Design Patterns and Principles/FactoryMethodPatternExample/FactoryMethodTest.cs
using System;

namespace FactoryMethodPatternExample
{
    class FactoryMethodTest
    {
        static void Main(string[] args)
        {
            Console.WriteLine("===== Factory Method Pattern Demonstration =====\n");

            DocumentFactory wordFactory = new WordDocumentFactory();
            Document word = wordFactory.CreateDocument();
            word.Open();

            DocumentFactory pdfFactory = new PdfDocumentFactory();
            Document pdf = pdfFactory.CreateDocument();
            pdf.Open();

            DocumentFactory excelFactory = new ExcelDocumentFactory();
            Document excel = excelFactory.CreateDocument();
            excel.Open();

            Console.WriteLine("\nAll document objects were created successfully using the Factory Method Pattern.");
        }
    }
}

File: Engineering concepts/FactoryMethodPatternExample/FactoryMethodTest.java
public class FactoryMethodTest {

    public static void main(String[] args) {

        System.out.println("===== Factory Method Pattern Demonstration =====\n");

        DocumentFactory wordFactory = new WordDocumentFactory();
        Document word = wordFactory.createDocument();
        word.open();

        DocumentFactory pdfFactory = new PdfDocumentFactory();
        Document pdf = pdfFactory.createDocument();
        pdf.open();

        DocumentFactory excelFactory = new ExcelDocumentFactory();
        Document excel = excelFactory.createDocument();
        excel.open();

        System.out.println("\nAll document objects were created successfully using the Factory Method Pattern.");
    }
}